test/parallel/test-child-process-cwd.js

```diff
@@ -66,10 +68,27 @@ function testCwd(options, expectPidType, expectCode = 0, expectData) {
     }));
 }
 
+{
+  assert.throws(() => {
+    testCwd({
+      cwd: new URL(`http://${tmpdir.path}`),
+    }, 'number', 0, tmpdir.path);
+  }, /The URL must be of scheme file/);
+
+  if (process.platform !== 'win32') {
+    assert.throws(() => {
+      testCwd({
+        cwd: new URL(`file://host${tmpdir.path}`),
+      }, 'number', 0, tmpdir.path);
+    }, /File URL host must be "localhost" or empty on/);
+  }
+}
+
 // Assume these exist, and 'pwd' gives us the right directory back
 testCwd({ cwd: tmpdir.path }, 'number', 0, tmpdir.path);
 const shouldExistDir = common.isWindows ? process.env.windir : '/dev';
 testCwd({ cwd: shouldExistDir }, 'number', 0, shouldExistDir);
+testCwd({ cwd: new URL(`file://${tmpdir.path}`) }, 'number', 0, tmpdir.path);
```

Comment:
There's no guarantee that `tmpdir.path` would make a valid URL:
```suggestion
testCwd({ cwd: pathToFileURL(tmpdir.path) }, 'number', 0, tmpdir.path);
```

Reply:
Hmmmm, because `tmpdir.path` used blow for a valid test case above:

```js
testCwd({ cwd: tmpdir.path }, 'number', 0, tmpdir.path);
```

Reply:
But it's used as a path here, not as a URL portion. It could be an issue if `tmpdir.path` contains special characters: https://github.com/nodejs/node/blob/bbab59d7ca7d4d49965940f67df6a0087cf0fd82/doc/api/url.md#L1122-L1126

Reply:
done